src/main/scala/tech/cryptonomic/conseil/tezos/michelson/renderer/MichelsonRenderer.scala

```diff
@@ -7,15 +7,20 @@ object MichelsonRenderer {
 
   implicit class MichelsonElementRenderer(val self: MichelsonElement) {
     def render(): String = self match {
+
+      // instructions
       case MichelsonSimpleInstruction(prim, List()) => prim
       case MichelsonSimpleInstruction(prim, args) => s"$prim ${args.map(_.render()).mkString(" ")}"
-      case MichelsonComplexInstruction(prim, args) => s"$prim ${args.render()}"
+      case MichelsonComplexInstruction(prim, args) => s"$prim ${args.map(_.render()).mkString(" ")}"
       case MichelsonInstructionSequence(args) => s"{ ${args.map(_.render()).mkString(" ; ")} }"
+      case MichelsonEmptyInstruction => "{}"
 
+      // expressions
       case MichelsonType(name, List()) => name
       case MichelsonType(name, args) => s"($name ${args.map(_.render()).mkString(" ")})"
       case MichelsonIntConstant(constant) => constant.toString
-      case MichelsonStringConstant(constant) => constant
+      case MichelsonStringConstant(constant) => "\"%s\"".format(constant)
```

Comment:
You can alternatively use "raw strings" when you need to embed quoted values, as in
`s""""$constant""""`

It's debatable which is more easy to read, yet I argue that the latter is harder to write incorrectly, because the compiler might catch non-matching quotes, or an incorrect interpolation...
With the `"...".format()` version, you'll find most errors at runtime